src/display/annotation_layer.js

```diff
@@ -67,7 +67,16 @@ AnnotationElementFactory.prototype =
         return new TextAnnotationElement(parameters);
 
       case AnnotationType.WIDGET:
-        return new WidgetAnnotationElement(parameters);
+        switch (parameters.data.fieldType) {
+          case 'Tx':
+            return new TextWidgetAnnotationElement(parameters);
+          default:
+            warn('Unimplemented Widget annotation type: ' +
+                  parameters.data.fieldType);
+            // Fallback to default one, which does not render anything
```

Comment:
Perhaps, but even so it seems unnecessarily complex. If that is the case, this should work too:
```diff
-  return new WidgetAnnotationElement(parameters);
+  switch (parameters.data.fieldType) {
+    case  'Tx':
+      return new TextWidgetAnnotationElement(parameters);
+    // case 'whatever':
+    // return new SomeMethodHere(parameters);
+  }
+  warn('Unimplemented Widget annotation type: ' +
+       parameters.data.fieldType);
+  // Fallback to default one, which does not render anything.
+  return new WidgetAnnotationElement(parameters);
```